File: protocol/src/test/ecmascript/zfooes/packet/EmptyObject.mjs

class EmptyObject {
    

    static PROTOCOL_ID = 0;

    protocolId() {
        return EmptyObject.PROTOCOL_ID;
    }

    static write(buffer, packet) {
        if (packet === null) {
            buffer.writeInt(0);
            return;
        }
        buffer.writeInt(-1);
    }

    static read(buffer) {
        const length = buffer.readInt();
        if (length === 0) {
            return null;
        }
        const beforeReadIndex = buffer.getReadOffset();
        const packet = new EmptyObject();
        
        if (length > 0) {
            buffer.setReadOffset(beforeReadIndex + length);
        }
        return packet;
    }
}
export default EmptyObject;

File: protocol/src/test/swift/zfooswift/packet/EmptyObject.swift
import Foundation


class EmptyObject : IProtocol {
    
    
    func protocolId() -> Int {
        return 0
    }
}

class EmptyObjectRegistration : IProtocolRegistration {
    func write(_ buffer: ByteBuffer, _ packet: Any?) {
        if (packet == nil) {
            buffer.writeInt(0)
            return
        }
        let message = packet as! EmptyObject
        buffer.writeInt(-1)
    }
    
    func read(_ buffer: ByteBuffer) -> Any {
        let length = buffer.readInt()
        let packet = EmptyObject()
        if (length == 0) {
            return packet
        }
        let beforeReadIndex = buffer.getReadOffset()
        
        if (length > 0) {
            buffer.setReadOffset(beforeReadIndex + length)
        }
        return packet
    }
}